### PR TITLE
implement entity, repository, cli, service packages

### PR DESCRIPTION
**Pull request by kianooshaz**

### CODE REVIEW

File: entity/user.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		Username  string
+		Email     string
+		Phone     string
+		Firstname string
+		Lastname  string
+		BirthDate time.Time
+		AboutMe   string
+		Gender    string
+		Links     []string
+		Avatar    string
```
Comment (sina-devel) [inline entity/user.go:17]: why the `User` entity has `Email`, `Phone` and etc fields?

File: entity/uuid.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package entity
+
+import "time"
+
+type UUID struct {
+	ID        uint
+	Value     string
+	CreatedAt time.Time
+	DeletedAt *time.Time
+}
```
Comment (sina-devel) [inline entity/uuid.go:10]: Do we need the `UUID` entity?

File: repository/repository.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package repository
+
+import "github.com/persiangophers/userland/entity"
+
+type (
+	Cash interface {
```
Comment (sina-devel) [inline repository/repository.go:6]: I think the name of this interface should be `Cache`.

File: repository/repository.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		DeleteProfile(id uint) error
+		HardDeleteProfile(id uint) error
+
+		// Password
+		CreatePassword(password entity.Password) error
+		GetPasswordByUUID(uuid string) (entity.Password, error)
+		GetPassword(id uint) (entity.Password, error)
+		DeletePassword(id uint) error
+		HardDeletePassword(id uint) error
+	}
```
Comment (sina-devel) [inline repository/repository.go:60]: It is better for each entity to have a separate interface so that it can be decoupled.

**Review by sina-devel — CHANGES_REQUESTED** [review]

File: entity/uuid.go (hunk 2 above)
Comment (kianooshaz) [reply entity/uuid.go:10]: i want know when created this uuid

File: repository/repository.go (hunk 3 above)
Comment (kianooshaz) [reply repository/repository.go]: yes its my mistake

File: repository/repository.go (hunk 4 above)
Comment (kianooshaz) [reply repository/repository.go:60]: Maybe, I will think on it

File: entity/user.go (hunk 1 above)
Comment (kianooshaz) [reply entity/user.go:17]: i need this struct to cache user

**Pull request closed without merging**